### PR TITLE
feat(hogql): property type info

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

Followup to https://github.com/PostHog/posthog/pull/16134

- Adds property and field types. 
- Makes sure we only show the relevant quick fixes

## How did you test this code?

See screengrab. Added tests for metadata.

### CODE REVIEW

File: posthog/hogql/functions/cohort.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def cohort(node: ast.Expr, args: List[ast.Expr], context: HogQLContext) -> ast.E
                 HogQLNotice(
                     start=arg.start,
                     end=arg.end,
-                    message=f"Searching for cohort by name. Replace with numeric id {cohorts[0][0]} to protect against renames.",
+                    message=f"Searching for cohort by name. Replace with numeric id {cohorts[0][0]} to protect against renaming.",
```
Comment (Twixes) [inline posthog/hogql/functions/cohort.py:42]:
```suggestion
                    message=f"Searching for cohort by name. Replace with numeric ID {cohorts[0][0]} to protect against renaming.",
```

File: posthog/hogql/test/test_metadata.py [hunk 2]
```diff
@@ -105,3 +108,61 @@ def test_metadata_expr_resolve_error(self):
...
+            | {
+                "isValid": True,
+                "inputExpr": None,
+                "inputSelect": query,
+                "notices": [
+                    {
+                        "message": f"Cohort #{cohort.pk} can also be specified as '{cohort.name}'",
+                        "start": 55,
+                        "end": 55 + len(str(cohort.pk)),
+                        "fix": f"'{cohort.name}'",
```
Comment (Twixes) [inline posthog/hogql/test/test_metadata.py:130]: I would think a fix should be one way, and this "fix" can be "fixed" by applying the other one ('Replace with numeric ID"), which can go on forever. 🤔 It's very useful to show the cohort name in a tooltip, but I suggest we don't have replacing that with the name as a fix, given the potential problems with renaming
BTW all the other notices end with a period, and this one doesn't. Given these are kind of linter hints, I would find them most natural if they were all single-sentence without a period

**Review by Twixes — COMMENTED** [review]
Very practical, just a few style comments

File: posthog/hogql/test/test_metadata.py (hunk 2 above)
Comment (mariusandra) [reply posthog/hogql/test/test_metadata.py:136]: - I changed a bunch of notices and smaller errors (that'll also end up in notices/errors here) to remove periods in the end.
- Why I included both actions: I think it's not right to assume what the user wants. They might actually want to use strings, as they have some system of naming cohorts they're using to make their queries easier to write/understand. I don't know and can't assume, but know I'd get frustrated when I'd have to copy/paste the text from the tooltip into the editor. So I left it in 🤷

**Comment by mariusandra** [comment]
Ready for a re-review. I also added support for showing types of regular fields like `timestamp`, not just properties.

File: frontend/src/queries/nodes/HogQLQuery/HogQLQueryEditor.tsx [hunk 3]
```diff
@@ -77,32 +77,54 @@ export function HogQLQueryEditor(props: HogQLQueryEditorProps): JSX.Element {
...
-                                                            diagnostics: [marker],
-                                                            kind: 'quickfix',
-                                                            edit: {
-                                                                edits: [
-                                                                    {
-                                                                        resource: model.uri,
-                                                                        textEdit: {
-                                                                            range: marker,
-                                                                            text: marker.hogQLFix ?? 'Error',
+                                                // Moncao gives us a list of markers that we're looking at, but without the quick fixes.
```
Comment (thmsobrmlr) [inline frontend/src/queries/nodes/HogQLQuery/HogQLQueryEditor.tsx:82]:
```suggestion
                                                // Monaco gives us a list of markers that we're looking at, but without the quick fixes.
```

**Review by thmsobrmlr — APPROVED** [review]

**Pull request merged**